examples/demo-app/src/utils/cloud-providers/carto.js

```diff
@@ -124,11 +124,18 @@ function loadMap(queryParams) {
     dispatch(setLoadingMapStatus(true));
     carto.PublicStorageReader.getVisualization(username, mapId).then((result) => {
       // These are the options required for the action. For now, all datasets that come from CARTO are CSV
-      const options = result.datasets.map((dataset) => ({
+      const options = result.datasets.map((dataset) => {
+        const datasetId = dataset.name.split('keplergl_public_v0_')[1];
+
         // TODO: customStorage should return dataset name without prefix
```

Comment:
Better move this `split` below the `TODO` comment

Reply:
I'm ok with this now, but this should be a public method on the custom storage, based on 'namespace' & 'version' parameters logic. That detail should be hidden away from kepler.gl